rosidl_generator_cpp/test/test_array_generator.hpp

```diff
@@ -0,0 +1,92 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TEST_ARRAY_GENERATOR_HPP_
+#define TEST_ARRAY_GENERATOR_HPP_
+
+#include <climits>
+#include <random>
+
+/**
+ * Helper function to generate a test pattern for boolean type.
+ * Alternating true (even index) and false (odd index) pattern.
+ * @param C Container (vector, array, etc) to be filled
+ * @param size How many elements to fill in. Must size<=container_size
+ */
+template<typename C>
+typename std::enable_if<std::is_same<typename C::value_type, bool>::value>::type
+test_vector_fill(C & container, size_t size, bool val1 = true, bool val2 = false)
+{
+  for (int i = 0; i < size; i++) {
+    if ((i % 2) == 0) {
+      container[i] = val2;
+    } else {
+      container[i] = val1;
+    }
+  }
+}
+
+/**
+ * Helper function to generate a test pattern for integer number types.
+ * The template type parameter must be an integer number type.
+ * Mininum and maximum values for the type and random values in the middle.
+ * @param C Container (vector, array, etc) to be filled
+ * @param size How many elements to fill in. Must size<=container_size
+ * @param min Minimum value in the rage to fill.
+ * @param max Maximum value in the rage to fill.
+ */
+template<typename C>
+typename std::enable_if<std::is_integral<typename C::value_type>::value &&
+~std::is_same<typename C::value_type, bool>::value>::type
```

Comment:
I think this should be like this:
```cpp
!std::is_same<typename C::value_type, bool>::value>::type

```

Also, I find this idiom slightly more readable, which we use in several places:

```cpp
template<typename C, typename std::enable_if<std::is_integral<typename C::value_type>::value &&
!std::is_same<typename C::value_type>::value>::type * = nullptr>
bool test_vector_fill(C & container, size_t size, typename C::value_type min, typename C::value_type max)

```

This way the template part doesn't pollute with the function signature.